internal/osutil/ping.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2015 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package osutil
+
+import (
+	"errors"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strconv"
+	"time"
+)
+
+var pattern = regexp.MustCompile(`time=(\d+\.?\d*)\s?([a-z]+)`)
+
+// Ping returns the latency to the given host
+// Raw sockets require root/administrator privileges, hence we work around
+// that by using the OS binary.
+func Ping(host string) (time.Duration, error) {
```

Comment:
Now replaced with TcpPing, which sadly leaves state around, but I guess that's the best we can do without dropping down to raw sockets.